examples/adaptive_tutorials/creating_modifying_adaptive_scripts.py

```diff
@@ -0,0 +1,388 @@
+# and :ref:`ref_SYNC_Curve_Flux_Barriers`.
+#
+# Set up example
+# --------------
+# Setting up this example consists of performing imports,
+# specifying the working directory, launching Motor-CAD,
+# and disabling all popup messages from Motor-CAD.
+#
+# Perform required imports
+# ~~~~~~~~~~~~~~~~~~~~~~~~
+
+import os
+from pathlib import Path
+import shutil
+import tempfile
+
+import ansys.motorcad.core as pymotorcad
+
+if "QT_API" in os.environ:
+    os.environ["QT_API"] = "pyqt"
+
+# %%
+# Import the ``geometry`` library. This is used to create Adaptive Templates.
+
+from ansys.motorcad.core.geometry import Arc, Coordinate, Line, rt_to_xy, xy_to_rt
+from ansys.motorcad.core.geometry_drawing import draw_regions
+
+# %%
+# Import the ``geometry_drawing`` library to use the geometry drawing feature
+
+
+# %%
+# Launch Motor-CAD
+# ~~~~~~~~~~~~~~~~~~~~~~~~~
+
+mc = pymotorcad.MotorCAD()
+
+# %%
+# Disable popup messages
+# ~~~~~~~~~~~~~~~~~~~~~~
+
+mc.set_variable("MessageDisplayState", 2)
+
+# %%
+# Open relevant file
+# ~~~~~~~~~~~~~~~~~~
+# Specify the working directory and open the relevant Synchronous Reluctance machine
+# template file for this example (i3).
+
+working_folder = Path(tempfile.gettempdir()) / "adaptive_example"
+try:
+    shutil.rmtree(working_folder)
+except:
+    pass
+mc.load_template("i3")
+mot_name = "Adaptive_Templates_Example_2"
+Path.mkdir(working_folder)
+mc.save_to_file(working_folder / (mot_name + ".mot"))
+
+# %%
+# Working on the adaptive templates example
+# -----------------------------------------
+# Work on the synchronous reluctance machine example motor
+# with curved flux barriers using adaptive templates
+# (:ref:`ref_adaptive_templates_example_2`).
+#
+# Adaptive templates should be disabled when working on a script
+# from an external IDE.
+# This allows the IDE to access the Standard Template Geometry.
+# Set the Geometry Templates Type to Standard (from Adaptive)
+
+mc.set_variable("GeometryTemplateType", 0)
+
+# %%
+# Instead of loading a script into Motor-CAD,
+# as done in the :ref:`ref_adaptive_templates_example_2` example,
+# the following script is run externally.
+# The following script is very similar to that in the
+# :ref:`ref_adaptive_templates_example_2` example,
+# but uses the geometry drawing feature to plot the geometry regions.
+#
+# Adaptive_Templates_Example_2_Debug.py script
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Adaptive Templates script to alter SYNCREL U-Shape rotor template to use curved rotor pockets.
+# This script does not support:
+# Zero inner/outer layer thickness
+# Inner/outer posts
+
+
+def get_barrier_centre_and_radius(coordinate_1, coordinate_2, coordinate_3, arc_direction):
+    """Calculate barrier arc centre and radius coordinates.
+
+    Parameters
+    ----------
+    coordinate_1 : Coordinate
+        Arc start coordinate.
+
+    coordinate_2 : Coordinate
+        Extra coordinate on arc
```

Comment:
It seems the coordinate_2 is end coordinate on the Arc. and coordinate_3 is extra or mid point on the arc